proposals/nnnn-cloud-app-transport-adapter.md

```diff
@@ -0,0 +1,243 @@
+5. If the web app recognizes the VIN it continues with its normal app behavior.
+6. If the web app does not recognize the VIN, the web app must send a show with the following message:
+```
+Activate Your Device
+Code: 12345
+Visit www.appwebsite.com/activate
+```
+7. The user must go to the applications website and register their connection before the web app can be used.
+
+This method of authentication uses a similar flow for activating streaming services (ie Roku/Apple TV -> Netflix)
+
+![alt text](../assets/proposals/nnnn-cloud-app-transport-adapter/web_app_high_level.png "Logo Title Text 1")
+
+### Obtaining Web App IP Address and Port
+Maintaining a list of urls for each web app will be similar to how policies for SDL applications are currently managed. There will need to be two new fields introduced in the app_policies section of the policy table. 
+
+This is what the app_policies section of a policy table that includes a web app endpoint might look like: 
+
+```JSON
+        "app_policies": {
+            "default": {
+                "keep_context": false,
+                "steal_focus": false,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4"]
+            },
+            "IOSApp12345": {
+                "keep_context": true,
+                "steal_focus": true,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4", "Location-1", "Emergency-1"]
+            },
+            "WebAppID12345" : {
+                "keep_context": true,
+                "steal_focus": true,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4"],
++                "endpoint": "https://10.0.0.1:8080",
++                "certificate" : "-----BEGIN CERTIFICATE-----\n" ...
+            }
+        }
+```
+
+The endpoint field includes the URL/IP and port that the SDL web app can be connected to. The certificate field is used for secured RPC service connections which equates to opening a secured websocket connection. The TLS "certificate" field could be optional to allow for unsecured RPC service connections.
+
+Endpoints can be added, removed, or updated via policy table updates. This means OEMs will be in control of the web apps that their head units attempt to connect to. 
+
+#### Transport Adapter Connection Flows
+
+- Ignition On: Core will read from the policy table for known connection endpoints. This information will be forwarded to the transport adapter level for client websocket connections to be opened.
+
+![alt text](../assets/proposals/nnnn-cloud-app-transport-adapter/web_app_ign_on.png "Logo Title Text 1")
+
+- Policy Table Update: After receiving a policy table updated and endpoint data has been updated, Core will open new connections for newly added endpoints and close connections for endpoints that have been "revoked"
```

Comment:
Should this be "After receiving a policy table update..." or "After receiving notification that the policy table has been updated..."?